server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -676,8 +679,14 @@ protected Symbol visitSubscriptExpression(SubscriptExpression node, ExpressionAn
                     }
                     try {
                         Symbol base = fieldProvider.resolveField(qualifiedName, List.of(), operation);
-                        if (base instanceof Reference) {
-                            throw e;
+                        if (base instanceof Reference baseRef) {
+                            if (baseRef.valueType().id() == ObjectType.ID) {
+                                if (baseRef.columnPolicy() == ColumnPolicy.STRICT || context.errorOnUnknownObjectKey()) {
+                                    throw new ObjectKeyUnknownException(e.getMessage());
+                                }
+                            } else {
+                                throw e;
```

Comment:
What do you think of passing in the session setting to `resolveField` and further down to `getField` and further to `DocTableInfo.getDynamic` - otherwise this is kinda re-implementing columnPolicy specific logic. I think we should keep that encapsulated in a single place.

Reply:
Initially I tried to create `Operation.READ_DYNAMIC` to indirectly pass down the session setting, mainly to prevent expanding the parameter list. How would you evaluate the impact of adding an extra parameter or an operation type?

Reply:
I see that these lines resembled what is in `getDynamic`, now there are some changes to the design that it actually returns `Literal.NULL`. Please review again.